odoo/addons/base/models/assetsbundle.py

```diff
@@ -322,13 +342,71 @@ def save_attachment(self, type, content):
 
         return attachment
 
-    def js(self):
-        attachments = self.get_attachments('js')
-        if not attachments:
-            content = ';\n'.join(asset.minify() for asset in self.javascripts)
-            return self.save_attachment('js', content)
+    def js(self, is_minified=True, with_sourcemap=False, name_unique=True):
+        url = None if name_unique else "/%s.js" % self.name
+        attachments = self.get_attachments('js', is_minified=is_minified, url_pattern=url)
+        if not attachments or (not name_unique and bundle_checksum.get(self.name, None) != self.checksum):
```

Comment:
```suggestion
        if not (attachments and name_unique or bundle_checksum.get(self.name) == self.checksum):
```
or so my editor tells me, and `None` is the default default of `dict.get` so that's redundant.

Alternatively, invert the entire thing and remove the not:

```python
if attachments and name_unique or bundle_checksum.get(self.name) == self.checksum):
    return attachments[0]

if not name_unique:
    bundle_checksum[self.name] = self.checksum
if with_sourcemap:
    return self.js_with_sourcemap(name_unique=name_unique)
else:
    content = ';\n'.join(asset.minify() if is_minified else asset.with_header(asset.content, minimal=False)
                         for asset in self.javascripts)
    return self.save_attachment('js', content, is_minified=is_minified, url=url)
```